### DISCUSSION TITLE
Secure Winlogbeats Question

### DISCUSSION THREAD

**Discussion by MrDigitalDick** [Security-Onion-Solutions/securityonion#8431] [Unsupported Versions]
Hi,

We have laptops as well as PC's and was wondering what the best way is to configure a secure connection back to the Security Onion node. Obviously if I have devices that are out in the field, how would they be able to send the logs back to SO? Should I use SSL, or API, I'm still learning all of this and trying to get my head around it all. Any help gratefully received.

Thanks,
Rich

**Comment by reyesj2** [comment]
If you're looking to gather logs from end devices you may want to look into Wazuh or Osquery both are baked into Security Onion.
You can download a custom agent for both from your SOC under "downloads"

**Reply by MrDigitalDick** [reply]
Hi, thanks for your reply. The question im asking is, how do i securely send the logs from the client to the SO node over internet?

Thanks,
Rich

**Reply by reyesj2** [reply]
In that case Wazuh handles the encryption of data. [Wazuh docs](https://documentation.wazuh.com/current/getting-started/components/wazuh-agent.html#communication-with-wazuh-server)

> The communication of the agent with the server takes place through a secure channel (TCP or UDP), providing data encryption and compression in real time. Additionally, it includes flow control mechanisms to avoid flooding, queueing events when necessary, and protecting the network bandwidth.

>You need to enroll the agent before connecting it to the server for the first time. This process provides the agent with a unique key used for authentication and data encryption.

**Reply by MrDigitalDick** [reply]
Ok, so I've managed to get one up and running using Wazuh. My next question is then, in your opinion what's the main difference between Wazuh and Winlogbeat? I've previously used Winlogbeat and Sysmon to send Windows event logs to SO, does Wazuh do the same thing out of the box or do i need to make some amendment?

Again, thank you so much for all your help so far!

Regards

**Reply by dougburks** [reply]
From https://docs.securityonion.net/en/2.3/sysmon.html#transport:
> Sysmon logs can be collected and transported using [Beats](https://docs.securityonion.net/en/2.3/beats.html#beats), [osquery](https://docs.securityonion.net/en/2.3/osquery.html#osquery), or [Wazuh](https://docs.securityonion.net/en/2.3/wazuh.html#wazuh).

Out of the box, the Wazuh agent for Windows probably just transports the standard Windows event logs.  If you want to transport Sysmon logs, then you may need to configure Wazuh to do so.
